docs/content/guide/compiler.ngdoc

```diff
@@ -380,3 +380,110 @@ restrict: 'E',
 replace: true
 ```
 
+### Double Compilation, and how to avoid it
+
+Double compilation occurs when an already compiled part of the DOM gets compiled again. This is an
+undesired effect and can lead to misbehaving directives, performance issues, and memory
+leaks.
+A common scenario where this happens is a directive that calls `$compile` in a directive link
+function on the directive element. In the following example, a directive adds a mouseover behavior
+to a button with `ngClick` on it:
+
+```
+angular.module('app').directive('addMouseover', function($compile) {
```

Comment:
I tried putting a div around it, but then the code blocks aren't rendered correctly. Seems to be an issue with the marked library: https://github.com/chjj/marked/issues/596